src/Web/Sfa.Das.Sas.Shared.Components/Orchestrators/ApprenticeshipOrchestrator.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using MediatR;
+using Sfa.Das.Sas.ApplicationServices.Models;
+using Sfa.Das.Sas.ApplicationServices.Queries;
+using Sfa.Das.Sas.ApplicationServices.Responses;
+using Sfa.Das.Sas.Shared.Components.Mapping;
+using Sfa.Das.Sas.Shared.Components.ViewComponents.ApprenticeshipDetails;
+using SFA.DAS.NLog.Logger;
+
+namespace Sfa.Das.Sas.Shared.Components.Orchestrators
+{
+    public interface IApprenticeshipOrchestrator
+    {
+        Task<FrameworkDetailsViewModel> GetFramework(string id);
+        ApprenticeshipType GetApprenticeshipType(string id);
+    }
+
+    public class ApprenticeshipOrchestrator : IApprenticeshipOrchestrator
+    {
+        private readonly IMediator _mediator;
+        private readonly ILog _logger;
+        private readonly IFrameworkDetailsViewModelMapper _frameworkDetailsViewModelMapper;
+
+        public ApprenticeshipOrchestrator(IMediator mediator, ILog logger, IFrameworkDetailsViewModelMapper frameworkDetailsViewModelMapper)
+        {
+            _mediator = mediator;
+            _logger = logger;
+            _frameworkDetailsViewModelMapper = frameworkDetailsViewModelMapper;
+        }
+
+        public async Task<FrameworkDetailsViewModel> GetFramework(string id)
+        {
+            var response = await _mediator.Send(new GetFrameworkQuery { Id = id });
+
+            string message;
+
+            switch (response.StatusCode)
+            {
+                case GetFrameworkResponse.ResponseCodes.InvalidFrameworkId:
```

Comment:
Should we be doing anything with these exceptions. I assume all will just bubble up as unhandled exceptions in the containing application?

Reply:
thats the idea yeah, the controller could handle these differently if we decide there is a need to